lib/drivers/weitai/weiTaiUSB.js

```diff
@@ -0,0 +1,1613 @@
+    try {
+      this.usbDevice.iface.release(true, () => {
+        this.usbDevice.device.close();
+        cb();
+      });
+    } catch (err) {
+      return cb(err, null);
+    }
+  }
+}
+
+module.exports = (config) => {
+  const cfg = _.clone(config);
+  _.assign(cfg.deviceInfo, {
+    tags: ['bgm'],
+    manufacturers: ['Weitai'],
+  });
+  const driver = new WeitaiUSB(cfg);
+
+  // With no date & time settings changes available,
+  // timezone is applied across-the-board
+  cfg.tzoUtil = new TZOUtil(cfg.timezone, new Date().toISOString(), []);
+
+  return {
+    /* eslint no-param-reassign:
+       [ "error", { "props": true, "ignorePropertyModificationsFor": ["data"] } ] */
+
+    detect(deviceInfo, cb) {
+      debug('no detect function needed', deviceInfo);
+      cb(null, deviceInfo);
+    },
+
+    setup(deviceInfo, progress, cb) {
+      debug('in setup!');
+      progress(100);
+      cb(null, { deviceInfo });
+    },
+
+    connect(progress, data, cb) {
+      debug('in connect!');
+      driver.openDevice(data.deviceInfo, (err) => {
+        if (err) {
+          data.disconnect = true;
+          debug('Error:', err);
+          return cb(err, null);
+        }
+        return cb(null, data);
+      });
+    },
+
+    getConfigInfo(progress, data, cb) {
+      debug('in getConfigInfo', data);
+      progress(0);
+      let _this = this;
+      (async () => {
+        //start
+        const result = await driver.getConfig(data);
+        data.deviceDetails = result;
+        cfg.deviceInfo.deviceId = 'MTM-I-00000'; // TODO: serial number
+        cb(null, data);
+      })().catch((error) => {
+        debug('Error in getConfigInfo: ', error);
+        cb(error, null);
+      });
+    },
+
+    buildBasalSchedules(settings) {
+      //carbRatio
+      let choList = settings.defaultCho;
+      let carbRatio = [];
+      for (let c = 0; c < choList.length; c++) {
+        if (c == 0) {
+          carbRatio.push({
+            amount: choList[c] / 10,
+            start: 0,
+          });
+        } else {
+          if (choList[c] != choList[c - 1] && c < 48) {
+            carbRatio.push({
+              amount: choList[c] / 10,
+              start: c * 30 * 60 * 1000,
+            });
+          }
+        }
+      }
+
+      //insulinSensitivity
+      let senList = settings.sensitiveSilver;
+      let insulinSensitivity = [];
+      for (let c = 0; c < senList.length; c++) {
+        if (c == 0) {
+          insulinSensitivity.push({
+            amount: choList[c] / 10,
+            start: 0,
+          });
+        } else {
+          if (senList[c] != senList[c - 1] && c < 48) {
+            insulinSensitivity.push({
+              amount: senList[c] / 10,
+              start: c * 30 * 60 * 1000,
+            });
+          }
+        }
+      }
+
+      //insulinSensitivity
+      let bgList = settings.lowBg;
+      let bgHighList = settings.highBg;
+      let bgTarget = [];
+      for (let c = 0; c < bgList.length; c++) {
```

Comment:
Please group time ranges with the same value together, as it would be displayed on the PDA. Currently it will appear as follows in the pump settings, when only a single BG target has been set on the pump:
![image](https://user-images.githubusercontent.com/418233/107377144-8d420780-6ae2-11eb-8799-aa8833eedf24.png)

Please do the same for basal schedules.